src/tracer.cc

```diff
@@ -279,34 +300,101 @@ TraceManager::SampleTrace(const std::string& model_name)
 
 TraceManager::Trace::~Trace()
 {
-  // Write trace now
-  setting_->WriteTrace(streams_);
+  if (setting_->mode_ == TRACE_MODE_TRITON) {
+    // Write trace now
+    setting_->WriteTrace(streams_);
+  } else if (setting_->mode_ == TRACE_MODE_OPENTELEMETRY) {
+#ifndef _WIN32
+    this->EndSpan();
+#else
+    LOG_ERROR << "Unsupported trace mode: "
+              << TraceManager::InferenceTraceModeString(setting_->mode_);
+#endif
+  }
 }
 
 void
 TraceManager::Trace::CaptureTimestamp(
     const std::string& name, uint64_t timestamp_ns)
 {
   if (setting_->level_ & TRITONSERVER_TRACE_LEVEL_TIMESTAMPS) {
-    std::lock_guard<std::mutex> lk(mtx_);
-    std::stringstream* ss = nullptr;
-    {
-      if (streams_.find(trace_id_) == streams_.end()) {
-        std::unique_ptr<std::stringstream> stream(new std::stringstream());
-        ss = stream.get();
-        streams_.emplace(trace_id_, std::move(stream));
-      } else {
-        ss = streams_[trace_id_].get();
-        // If the string stream is not newly created, add "," as there is
-        // already content in the string stream
-        *ss << ",";
+    if (setting_->mode_ == TRACE_MODE_TRITON) {
+      std::lock_guard<std::mutex> lk(mtx_);
+      std::stringstream* ss = nullptr;
+      {
+        if (streams_.find(trace_id_) == streams_.end()) {
+          std::unique_ptr<std::stringstream> stream(new std::stringstream());
+          ss = stream.get();
+          streams_.emplace(trace_id_, std::move(stream));
+        } else {
+          ss = streams_[trace_id_].get();
+          // If the string stream is not newly created, add "," as there is
+          // already content in the string stream
+          *ss << ",";
+        }
+      }
+      *ss << "{\"id\":" << trace_id_ << ",\"timestamps\":["
+          << "{\"name\":\"" << name << "\",\"ns\":" << timestamp_ns << "}]}";
+    } else if (setting_->mode_ == TRACE_MODE_OPENTELEMETRY) {
+#ifndef _WIN32
+      otel_common::SystemTimestamp otel_timestamp{
+          (time_offset_ + std::chrono::nanoseconds{timestamp_ns})};
+      if (trace_span_ == nullptr) {
+        this->InitSpan(otel_timestamp);
+      }
+      trace_span_->AddEvent(
+          name, otel_timestamp, {{"triton.steady_timestamp_ns", timestamp_ns}});
```

Comment:
Perhaps remove `steady_`?

Reply:
That is a reference to the steady clock

Reply:
I think it is an implementation detail that should be hidden from the user. Is there any way to make sure these timestamps have the same starting point?

Reply:
This way OpenTelemetry's spans contain the same info as our current traces (Triton trace API). 
 OpenTelemetry SDK relies on system's clock for event timestamps. Triton Tracing records timestamps using steady_clock. This is a monotonic clock, i.e. time is always moving forward. It is not related to wall clock time (for example, it can be time since last reboot). These 2 clocks are different clocks, and steady clock is usually used for more precise computations. However, so that traces are displayed in the trace collector correctly, system's clock is used. To sum up, since our current traces report steady timestamp only, I also include it in pan's info

Reply:
Also, I'm curious how this would work with some OTEL backends like prometheus and Jaeger. Have we done an end-to-end integration with these tools to see how the visualization would look like?

Reply:
@tanmayv25 we've discussed this at some point. Are we ok to remove triton's timestamp from trace's info?

@Tabrizian I've tested it with Jaeger, because this was the tool, that the customer is planning to use.  I showed it to Tanmay, it looks as expected. The introduction of `time_offset_` was driven by Jaeger's expectation for OpenTelemetry's timestamp to put trace on time axis.

Reply:
We can remove triton's timestamp from OTEL trace for now. We can document the difference in behavior of the two trace modes.

Reply:
Removed steady timestamp from OTel spans.

Reply:
@Tabrizian and @tanmayv25 I've removed steady timestamp and added  [entry](https://github.com/triton-inference-server/server/blob/197595f13d8c40aa0f4fb5b6a66ad08a540bf407/docs/user_guide/trace.md#differences-in-trace-contents-from-tritons-trace-output) to docs